stdlib/public/core/CommandLine.swift

```diff
@@ -117,6 +123,55 @@ extension CommandLine {
       _arguments = newValue
     }
   }
+
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS) || os(visionOS)
+@_extern(c, "_NSGetExecutablePath")
+@usableFromInline
+internal func _NSGetExecutablePath(
+  _ buf: UnsafeMutablePointer<CChar>,
+  _ bufsize: UnsafeMutablePointer<UInt32>
+) -> CInt
+#endif
+
+  /// The path to the current executable.
+  ///
+  /// - Important: On some systems, it is possible to move an executable file on
+  ///   disk while it is running. If the current executable file is moved, the
+  ///   value of this property is not updated to its new path.
+  @_unavailableInEmbedded
+  #if os(WASI)
+  @available(*, unavailable, message: "Unavailable on WASI")
+  #endif
+  @available(SwiftStdlib 6.3, *)
+  @_alwaysEmitIntoClient
+  public static let executablePath: String = {
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS) || os(visionOS)
+    // _NSGetExecutablePath() returns non-zero if the provided buffer is too
+    // small and updates its *bufsize argument to the required value, so we can
+    // just set a reasonable initial value, then loop and try again on failure.
+    var byteCount = UInt32(128)
+    while true {
```

Comment:
Not a fan of having no exit condition for this loop – is it guaranteed to only need a second loop at most? Can we use a `firstTime` Bool or something?

Reply:
It may loop more than once if the implementation of `_NSGetExecutablePath()` allows for the path to change at runtime and the executable is moved in the middle of this loop. However, that may just be pedantry.